src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -241,46 +226,35 @@ static bool equalMetadataValues(BSONObj& md1, BSONObj& md2)
 
 /* ****************************************************************************
 *
-* equalMetadataVectors -
+* equalMetadata -
 *
-* Given two vectors with the same metadata names-types, check that all the values
+* Given two metadata object assuming they have the same size (i.e. number of elements). check that all the values
 * are equal, returning false otherwise.
 *
 * Arguments are passed by reference to avoid "heavy" copies
 */
-static bool equalMetadataVectors(BSONObj& mdV1, BSONObj& mdV2)
+static bool equalMetadata(BSONObj& md1, BSONObj& md2)
```

Comment:
Could be a method in the Metadata class:

    if (md1.equals(md2))
    {
    }

Easier to find the function later if its in the class. Less prone for copies.
If you wish to keep all mongo stuff (BSONObj) encapsulated in mongBackend, a method AND a function would be necessary.